### PR TITLE
docs(ticdc): add documentation for running integration tests in docker

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve? 
part of https://github.com/pingcap/ticdc/issues/2751
close https://github.com/pingcap/ticdc/issues/3714

### What is changed and how it works?

add documentation for running integration tests in docker. 

[rendered](https://github.com/hi-rustin/ticdc/blob/rustin-patch-docs-docker/tests/integration_tests/README.md)
### Check List 

Tests 

 - No code

Code changes

None

Side effects

None

Related changes

None

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
/cc @overvenus @amyangfei @maxshuang 

Could you please take a look? Thanks! 💚 💙 💜 💛 ❤️

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -25,7 +25,8 @@ make
 
 ### Running tests
 
-This project contains unit tests and integration tests with coverage collection. See [tests/README.md](./tests/README.md) for how to execute and add tests.
+This project contains unit tests and integration tests with coverage collection.
```
Comment (maxshuang) [inline CONTRIBUTING.md:28]: Could u change the "Format of the Commit Message" in this pr?  
From:
```
<subsystem>: <what changed>
```
To:
```
<subsystem>(ticdc): <what changed>
```

File: tests/integration_tests/README.md [hunk 2]
```diff
@@ -1,52 +1,101 @@
...
-    > All Tiflash required files can be found in [tidb-community-server](https://download.pingcap.org/tidb-community-server-v4.0.2-linux-amd64.tar.gz) packages. You should put `flash_cluster_manager`, `libtiflash_proxy.so` and `tiflash` into `bin` directory in TiCDC code base.
+   > If you are running tests on MacOS, tidb related binaries can be downloaded from tiup mirrors, such as https://tiup-mirrors.pingcap.com/tidb-v4.0.2-darwin-amd64.tar.gz. And `sync_diff_inspector` can be compiled by yourself from source [tidb-tools](https://github.com/pingcap/tidb-tools)
+
+   > All Tiflash required files can be found in [tidb-community-server](https://download.pingcap.org/tidb-community-server-v4.0.2-linux-amd64.tar.gz) packages. You should put `flash_cluster_manager`, `libtiflash_proxy.so` and `tiflash` into `bin` directory in TiCDC code base.
 
 2. The following programs must be installed:
 
-    * `mysql` (the MySQL cli client, currently [mysql client 8.0 is not supported](https://github.com/pingcap/tidb/issues/14021))
+   * `mysql` (the MySQL cli client,
+     currently [mysql client 8.0 is not supported](https://github.com/pingcap/tidb/issues/14021))
```
Comment (maxshuang) [inline tests/integration_tests/README.md:29]: Could u add some normal way to install mysql client by yum/apt/brew? So I can install mysql client by one step(copy and paste).
like: 
```
sudo yum install mysql-community-client.x86_64 -y
```

File: tests/integration_tests/README.md [hunk 3]
```diff
@@ -1,52 +1,101 @@
...
+
+3. The user used to execute the tests must have permission to create the folder /tmp/tidb_cdc_test. All test artifacts
+   will be written into this folder.
+
+### Run integration tests in docker
+
+The following programs must be installed:
 
-3. The user used to execute the tests must have permission to create the folder /tmp/tidb_cdc_test. All test artifacts will be written into this folder.
+* [docker](https://www.docker.com/)
```
Comment (maxshuang) [inline tests/integration_tests/README.md:38]: Same as mysql client, is there any quick way to install these dependency by one step?

File: tests/integration_tests/README.md [hunk 4]
```diff
@@ -1,52 +1,101 @@
...
+#### Run integration tests in docker
+
+> **Warning:**
+> These scripts and files may not work under the arm architecture,
+> and we have not tested against it.
+> Also, we currently use the PingCAP intranet address in our download scripts,
+> so if you do not have access to the PingCAP intranet you will not be able to use these scripts.
+> We will try to resolve these issues as soon as possible.
+
+1. If you want to run kafka tests,
```
Comment (maxshuang) [inline tests/integration_tests/README.md:78]: Do we still need zookeeper to test Kafka here?

**Comment by maxshuang** [comment]
We get a very quick way to test Kafka and Mysql now. Exciting!

File: CONTRIBUTING.md (hunk 1 above)
Comment (hi-rustin) [reply CONTRIBUTING.md:28]: See: https://github.com/pingcap/ticdc/pull/3549

File: tests/integration_tests/README.md (hunk 4 above)
Comment (hi-rustin) [reply tests/integration_tests/README.md:78]: It is already in docker-compose. At the moment kraft is not production-ready, so we still need it.

File: tests/integration_tests/README.md (hunk 2 above)
Comment (hi-rustin) [reply tests/integration_tests/README.md:29]: I found it difficult to explain these commands in our documentation, so I used the official documentation link.

File: tests/integration_tests/README.md (hunk 3 above)
Comment (hi-rustin) [reply tests/integration_tests/README.md]: I used the official documentation link.

**Review by maxshuang — APPROVED** [review]

**Review by overvenus — APPROVED** [review]
LGTM

**Review by asddongmen — APPROVED** [review]

**Comment by asddongmen** [comment]
Bravo~！

**Pull request merged**